### PR TITLE
Upgrade to Chromium 59

### PR DESCRIPTION
**Pull request by alexeykuzmin**
Instead of #9766.

**Failing tests**:
- [ ] [Windows] BrowserWindow module "webPreferences" option "sandbox" option can print to PDF
- [ ] [Windows] BrowserWindow module window states resizable state works for a frameless window

_Both failing tests are disabled for now._

### CODE REVIEW

**Comment by alexeykuzmin** [comment]
On Windows I have a couple of dozens size/position related tests fail with all numbers being off by one or two pixels. Error messages are like this:
```
window.open
   ✕ does not override child options
    Assertion error: 'size: 352 451' == 'size: 350 450'
    ...
```

On Mac it doesn't happen, so I guess it has something to do with native window code.
Can someone please take a look?

**Comment by kevinsawicki** [comment]
> On Windows I have a couple of dozens size/position related tests fail with all numbers being off by one or two pixels.

What DPI are you using locally?

**Comment by kevinsawicki** [comment]
> What DPI are you using locally?

I was able to reproduce the failure at 150% zoom, but it passes for me at 100% zoom.

**Comment by alexeykuzmin** [comment]
> I was able to reproduce the failure at 150% zoom, but it passes for me at 100% zoom.

Zoom was set to 125%, with 100% those tests passed. Thanks.

**Comment by KagamiChan** [comment]
Seems failing tests related to Windows system zoom level are already reported in #9691 
As there're some other issues concerning zoom level, could we have tests for zoom level > 100% on Windows ci runner? Not for this chromium upgrade but someday after.

**Comment by gerhardberger** [comment]
@alexeykuzmin I fixed the OSR-related errors on macOS.

From the looks of it, there should be no Windows specific errors, but if there are, let me know and we might be able to solve it, then we can enable OSR in `update.py`.

**Comment by alexeykuzmin** [comment]
@gerhardberger Cool, thanks! Let's enable OSR and see what happens.

**Comment by alexeykuzmin** [comment]
> there should be no Windows specific errors, but if there are, let me know and we might be able to solve it

@gerhardberger compilation on Windows failed, check this out:
http://208.52.191.140:8080/job/electron-win-x64/3515/console

**Comment by alexeykuzmin** [comment]
@gerhardberger I pretty you've already seen it, but Windows build failed again.
http://208.52.191.140:8080/job/electron-win-x64/3516/console

**Comment by gerhardberger** [comment]
@alexeykuzmin yeah I saw it. I managed to resolve the errors all but one, because that is a different sort of issue.

Tomorrow I will take a closer look at it!

**Comment by alexeykuzmin** [comment]
A single test failed in a [Mac build](http://208.52.191.140:8080/job/electron-mas-x64/4540/console):
```
not ok 775 <webview> tag permission-request event emits when using navigator.getUserMedia api
  AssertionError: [ 'DevicesNotFoundError' ] deepEqual [ 'PermissionDeniedError' ]
      at HTMLElement.<anonymous> (/Users/electron/workspace/electron-mas-x64/spec/webview-spec.js:981:16)
      at WebViewImpl.dispatchEvent (/Users/electron/workspace/electron-mas-x64/out/D/Electron.app/Contents/Resources/electron.asar/renderer/web-view/web-view.js:175:22)
      at EventEmitter.<anonymous> (/Users/electron/workspace/electron-mas-x64/out/D/Electron.app/Contents/Resources/electron.asar/renderer/web-view/guest-view-internal.js:79:15)
      at emitThree (events.js:116:13)
      at EventEmitter.emit (events.js:197:7)
```

@deepak1556 , maybe you can take a look? I'm absolutely not familiar with media-stuff.

**Comment by alexeykuzmin** [comment]
@gerhardberger win builds fail again =(
http://208.52.191.140:8080/job/electron-win-x64/3527/console

**Comment by alexeykuzmin** [comment]
@gerhardberger
I have disabled OSR to run tests on Win. Please enable OSR back if/when you push any changes to fix compilation errors. Thanks!

**Comment by alexeykuzmin** [comment]
**TODO** Win builds failed:
```
[1319/1319] LINK_EMBED electron.exe
FAILED: C:\Python27\python.exe gyp-win-tool link-with-manifests environment.x64 True electron.exe "C:\Python27\python.exe gyp-win-tool link-wrapper environment.x64 False link.exe /nologo /OUT:electron.exe @electron.exe.rsp" 1 mt.exe rc.exe "obj\electron.electron.exe.intermediate.manifest" obj\electron.electron.exe.generated.manifest ..\..\atom\browser\resources\win\atom.manifest
LINK : fatal error LNK1104: cannot open file 'C:/e/workspace/electron-win-x64/vendor/download/libchromiumcontent/shared_library/fx_freetype.lib'
```

**Comment by setpixel** [comment]
I just wanted you guys to know, that I appreciate your hard work and I believe in you!

**Comment by gerhardberger** [comment]
@alexeykuzmin It seems the OSR-related errors resolved, only that last linking error is present.

**Comment by alexeykuzmin** [comment]
>  It seems the OSR-related errors resolved, only that last linking error is present.

@gerhardberger That's great, thanks!
I'll look at Windows build tomorrow. For some reason it works on my machine while it obviously shouldn't.

**Comment by alexeykuzmin** [comment]
@kevinsawicki 
http://208.52.191.140:8080/job/electron-osx-x64/4561/console :
> IOError: [Errno 28] No space left on device

Can we do something to prevent it from happening in the future? It's kind of inconvenient )

**Comment by kevinsawicki** [comment]
> Can we do something to prevent it from happening in the future? It's kind of inconvenient )

The machine has 500GB of space, the `libchromiumcontent-mas` build was taking up 400GB of it, with a `_bad_scm` folder taking up 278GB in it.

**Comment by kevinsawicki** [comment]
> Can we do something to prevent it from happening in the future? It's kind of inconvenient )


Added it to the clean option in https://github.com/electron/libchromiumcontent/pull/321

**Comment by johnryan** [comment]
@alexeykuzmin What version of Chromium does electron currently use? I stumbled on this issue #10000 which *SEEMS* to have been resolved here: https://chromium.googlesource.com/chromium/src/+/b39b8d6fb03d46dff5556e7a5c428ac9302f5579
I'm trying to figure out if this PR will resolve the issue.

THANKS!

**Comment by alexeykuzmin** [comment]
@johnryan 
The latest stable Electron v1.6.11 is based on Chromium 56, v1.7.4 (beta) – on Chromium 58.
The changes you are referring to have gotten into 59th release and will be part of Electron 1.8.*.

**Comment by johnryan** [comment]
@alexeykuzmin Awesome, thanks for the info...this is likely out of the scope of this PR, but do you know any timelines for that? Just trying to figure out how to deal with this issue with our users (or figure out a workaround). If that's not your area of expertise no worries, i'll keep digging elsewhere. Thanks again!

**Comment by alexeykuzmin** [comment]
@johnryan I have no idea when 1.8.0 is going to be released.
1.6.0 beta was published in February, 1.7.0 beta – in May, so you can roughly estimate 1.8.0 beta release date to be somewhere in August.

Unfortunately I also can't help you with a workaround.

**Comment by alexeykuzmin** [comment]
Rebased on the latest master to fix merge conflicts.

**Comment by alexeykuzmin** [comment]
Waiting for #10053 to be merged.

**Comment by kevinsawicki** [comment]
> Waiting for #10053 to be merged.

@alexeykuzmin just curious, why does this need to wait on #10053?

**Comment by alexeykuzmin** [comment]
> @alexeykuzmin just curious, why does this need to wait on #10053?

@alespergl told me that there is some sort of incompatibility between libcc and Electron build configurations right now. Maybe I didn't understand him correctly.

**Comment by alespergl** [comment]
The reason is that the debug mode is a two part change. Currently the libcc part remains merged to master while the Electron part was reverted due to the failing tests. Ideally, if this PR is not too critical, I would like to resolve the issues causing the tests to fail and re-merge the second part to Electron before merging this. I'm expecting to have it finished this week, but it's a bit time consuming due to the looong libcc builds (which, to make things worse, seem to be busted right now), so if this PR is a priority, reverting the debug mode commit from libcc is also an option.

**Comment by alexeykuzmin** [comment]
@kevinsawicki All the builds failed on libcc download with `urllib2.HTTPError: HTTP Error 403: Forbidden` error. What could go wrong? Is it because they were triggered by me and I'm not a GitHub employee?

**Comment by johnryan** [comment]
+1 For merging asap, hoping to resolve the native crashes i'm seeing in chromium 58 (#10000). Let me know if I can assist in any way. Thanks!

**Comment by kevinsawicki** [comment]
> All the builds failed on libcc download with urllib2.HTTPError: HTTP Error 403: Forbidden error. What could go wrong? Is it because they were triggered by me and I'm not a GitHub employee?

I don't think those builds are complete yet, there was a pretty deep queue for mac/windows builds that I think is still being churned through.

**Comment by alexeykuzmin** [comment]
> those builds are complete yet

Oh, yes, you're right. I saw a green checkmark and thought that all builds were successful =/
But there are only linux builds.

**Comment by kevinsawicki** [comment]
> Oh, yes, you're right. I saw a green checkmark and thought that all builds were successful =/
But there are only linux builds.

I'm monitor and make sure they get retriggered if they don't complete, there was an out of memory issue today on some of the jenkins windows workers so some builds failed because of that.

**Comment by kevinsawicki** [comment]
@alexeykuzmin looks like the builds are now failing on macOS and Windows with linker errors, is this something you are looking into? Or should I?

**Comment by gerhardberger** [comment]
@kevinsawicki seems like something went wrong with the linking of OSR code. I will take a look at it today.

**Comment by alexeykuzmin** [comment]
> looks like the builds are now failing on macOS and Windows with linker errors

@kevinsawicki It happened after the rebase, I think it's somehow related to `is_debug=true` being added to libcc build config. @alespergl, am I right?

**Comment by kevinsawicki** [comment]
>  It happened after the rebase, I think it's somehow related to is_debug=true being added to libcc build config. @alespergl, am I right?

Then perhaps we should rebase out the debug mode commits until this is resolved so we can move forward here and get this released to beta. Thoughts?

**Comment by alexeykuzmin** [comment]
Then we should probably revert electron/libchromiumcontent@7d889fde3b5c7083c554675c97142547d7605057 in master, otherwise we would get exactly the same errors after merge.

**Comment by alespergl** [comment]
Let's first confirm that it's related. I don't know what the error is (haven't looked yet), but it seems unlikely that debug mode would not link.

**Comment by alexeykuzmin** [comment]
> it seems unlikely that debug mode would not link

I assumed that only because there a lot of "debug" and "release" mentions in the logs )

**Comment by alespergl** [comment]
Are you guys talking about libcc or Electron build? If it's Electron than yes, that won't link to master libcc currently. If you give me maybe 4 more hours, I can make it work again.

**Comment by alexeykuzmin** [comment]
@alespergl Libcc is fine, we are talking about Electron builds (this PR).

**Comment by kevinsawicki** [comment]
> Are you guys talking about libcc or Electron build? If it's Electron, then yes, that won't link to master libcc currently

Yes, right now, this Electron branch does not link against the chrome 59 libcc branch.

**Comment by alexeykuzmin** [comment]
In a meanwhile we can revert debug mode in the libcc PR (electron/libchromiumcontent#313)
and run the builds, they will take about 4-5 hours to finish.

If by that time Electron build is fixed we will revert the revert (effectively restore libcc tree as it is now),
and if Electron build is not fixed, we will update a libcc commit Electron uses to a new one.

@kevinsawicki what do you think?

**Comment by kevinsawicki** [comment]
> what do you think?

Yeah, that sounds good, let me know once you revert in the libcc branch I'll trigger the linux libcc builds

**Comment by alexeykuzmin** [comment]
> let me know once you revert in the libcc branch

just pushed

**Comment by kevinsawicki** [comment]
@alexeykuzmin Was able to link this branch on Linux with the built libcc of https://github.com/electron/libchromiumcontent/commit/ec73dfbd6718c49a9fceb6f1672ba125082461a2 👍

**Comment by alexeykuzmin** [comment]
@alespergl Any updates on fixing the Electron build? libcc builds are almost completed, I'm going to push soon.

**Comment by alexeykuzmin** [comment]
@kevinsawicki I expect some tests to fail, especially on Linux. Can we disable them and merge, and fix later? (Of course if it's not something uberimportant.)

**Comment by alespergl** [comment]
@alexeykuzmin I had to do another iteration of libcc commit and build because the change that I originally thought would do the final trick didn't work as expected. Anyway I'm now in final stage of verification and if the tests pass now, I can then clean up my branch of work-in-progress commits and it's ready for merging. You can then rebase on top. I'll update you in a moment if things passed or not.

**Comment by alespergl** [comment]
So the changes are ready. In order to have both debug mode and chromium upgrade, I'm suggesting the following:
- merge https://github.com/electron/libchromiumcontent/pull/322
- merge https://github.com/electron/electron/pull/10053
- rebase chromium upgrade on top of the new tips of both libcc and Electron masters

Note: It can also be done in the opposite order (i.e. chromium59 first, debug mode second), but since debug mode is already partially merged (the commit Alexey reverted in his branch), the first option is less messy, IMO.

**Comment by alexeykuzmin** [comment]
> merge electron/libchromiumcontent#322
merge #10053

I agree, less mess.
Also builds in debug mode should be faster.

**Comment by alexeykuzmin** [comment]
For some reason only 137 tests [were run](http://208.52.191.140:8080/job/electron-win-x64/3611/console) on Windows when the build failed. Does it mean that the process crashed or something?

**Comment by kevinsawicki** [comment]
> For some reason only 137 tests were run on Windows when the build failed. Does it mean that the process crashed or something?

Yeah, that usually means a main process crash.

**Comment by kevinsawicki** [comment]
> I expect some tests to fail, especially on Linux. Can we disable them and merge, and fix later? (Of course if it's not something uberimportant.)

Tests are looking pretty good on Linux, all Linux builds are green as of https://github.com/electron/electron/pull/9946/commits/8dbf38b2a51abf5ecfa5d3a86186747a99030a9e 🍏 

I disabled the `beginFrameSubscription` specs due to a crash I was seeing but I only saw it happening from a docker container, not from a parallels ubuntu VM, so I don't think it is related to the upgrade.

**Comment by alespergl** [comment]
Debug mode PRs are merged. @alexeykuzmin you can rebase on new tip of master. 🙇

**Comment by steem** [comment]
If 59 is ready, will it support headless feature?

**Comment by alexeykuzmin** [comment]
> If 59 is ready, will it support headless feature?

It's not ready yet. See issue #228 for headless mode support.

**Comment by alexeykuzmin** [comment]
Rebased on the latest master to resolve merge conflicts.

**Comment by alexeykuzmin** [comment]
On Windows Electron crashes (?) while running tests.
All builds failed on random tests, probably it just fails after some time after launch.
I don't really know how investigate/fix it, can any one take a look? @kevinsawicki, @zcbenz, @alespergl 

Btw, can anyone start Linux tests?

**Comment by alexeykuzmin** [comment]
Linux builds fail on bootstrap.py execution.
E.g. https://travis-ci.org/electron/electron/jobs/256806091
```
Running bootstrap.py
Traceback (most recent call last):
  File "setup.py", line 37, in <module>
    from boto import __version__
  File "/home/travis/build/electron/electron/vendor/boto/boto/__init__.py", line 1216, in <module>
    boto.plugin.load_plugins(config)
  File "/home/travis/build/electron/electron/vendor/boto/boto/plugin.py", line 93, in load_plugins
    _import_module(file)
  File "/home/travis/build/electron/electron/vendor/boto/boto/plugin.py", line 75, in _import_module
    return imp.load_module(name, file, filename, data)
  File "/usr/lib/python2.7/dist-packages/google_compute_engine/boto/compute_auth.py", line 19, in <module>
    from google_compute_engine import logger
ImportError: No module named google_compute_engine
```
@zcbenz, maybe you can help?

**Comment by jkleinsc** [comment]
@alexeykuzmin I believe the Travis failures are due to Travis switching from Ubuntu Precise to Ubuntu Trusty:
https://blog.travis-ci.com/2017-07-11-trusty-as-default-linux-is-coming

The specific error you are seeing seems to be due to this:
https://github.com/travis-ci/travis-ci/issues/7940

I am going to change the travis setup to explicitly use Ubuntu Precise to see if that resolves the build failure on Travis.

**Comment by jkleinsc** [comment]
Ok it looks like Precise has its own set of issues.  I am going to try the Trusty fix suggested here:
https://github.com/travis-ci/travis-ci/issues/7940

**Comment by jkleinsc** [comment]
Applying the Trusty fix:
```yaml
before_install:
  - export BOTO_CONFIG=/dev/null
```
Fixed the bootstrap issue, but it appears there are still issue on Travis.
The arm build fails because of a conflict between the `gcc-multilib` package and the `g++-arm-linux-gnueabihf` package.

@zcbenz I would appreciate if you could look at this.  If not I will try to dig in further tomorrow.

**Comment by setpixel** [comment]
I know this aspect of this work can seem pretty thankless sometimes. I know I'm not speaking just for myself when I say that I appreciate everyone's hard work on this! I'm really looking forward to using 59 in electron. You guys are doing a great job and thanks! If you're in NYC anytime, beers on me! ❤️ 👍

**Comment by alexeykuzmin** [comment]
@alespergl these weird failures while running tests on Windows do not happen on Release build of Electron. And I don't remember them happening before Debug build became "true debug".
Could you please take a look what is going on there?

**Comment by alespergl** [comment]
I will have a look, yes.

**Comment by alexeykuzmin** [comment]
@alespergl FYI
I just built libcc and Electron from Ch59 branches with debug mode related changes reverted.
And now tests do not mysteriously fail. There are three failed tests though ), but it's much better than silent failures.

**Comment by alespergl** [comment]
@alexeykuzmin The Windows tests are fixed. There was a bug (introduced in this PR) where null was passed to `memcpy` and this bug was being flagged by the *debug* VC runtime. So thanks to the debug mode this bug did not get into master.

**Comment by alexeykuzmin** [comment]
Another forcepush to resolve merge conflicts.

**Comment by alexeykuzmin** [comment]
Although all Windows builds are green, only 439 tests were executed in each one. That's not ok.
I'm going to investigate in tomorrow.

**Comment by zcbenz** [comment]
I'm doing a force push to include some Linux fixes on master.

**Comment by alespergl** [comment]
> only 439 tests were executed

Note that this is also the case in master.

**Comment by alexeykuzmin** [comment]
> > only 439 tests were executed
> Note that this is also the case in master.

@zcbenz, maybe we can merge without fixing it?, if Mac and Linux tests are all green.

**Comment by zcbenz** [comment]
> @zcbenz, maybe we can merge without fixing it?, if Mac and Linux tests are all green.

While I'm good merging this, I'm not sure if we should do it after cutting a new 1.7.x release,  @jkleinsc @zeke.

**Comment by zcbenz** [comment]
I'm doing another force push for a few other tests related commits on master.

**Comment by kevinsawicki** [comment]
Upgraded this branch to use Node 8.2.1 via #9992

**Comment by alexeykuzmin** [comment]
> I'm doing a force push to include some Linux fixes on master.

It seems like builds still fails on Linux x64.

**Comment by alexeykuzmin** [comment]
> Upgraded this branch to use Node 8.2.1 via #9992 

@kevinsawicki 
Now on Windows linking fails in both Debug and Release builds.

**Comment by kevinsawicki** [comment]
> Now on Windows linking fails in both Debug and Release builds.

Thanks, taking a look

**Comment by zcbenz** [comment]
> It seems like builds still fail on Linux x64.

There are a few crashes on Linux and Windows when running tests, I think it is because using Debug build of libcc has revealed some memory corruptions in Electron. The Debug build of master branch is also crashing when running tests.

**Comment by alespergl** [comment]
> memory corruptions in Electron

Yes, I tried to find the source of those corruptions yesterday, but unfortunately I wasn't able to pin it down. I'm just using my forensic skills, but maybe @zcbenz could have a better idea with his insight into how things are implemented.

If we can't find the bug, I see these options for unblocking (since it's not a regression)
- disable the failing test on Linux
- disable the debugging checks in the `tcmalloc` component (same as I did for Skia, ANGLE, V8)
- run tests against the release build (with `test.py -c R`)

**Comment by zcbenz** [comment]
I'm rebasing this branch on master.

**Comment by alexeykuzmin** [comment]
Two tests failed in all Windows builds:
```
not ok 113 BrowserWindow module "webPreferences" option "sandbox" option can print to PDF
  AssertionError [ERR_ASSERTION]: { Error: Failed to generate PDF message: 'Failed to generate PDF', name: 'Error' } == null
not ok 153 BrowserWindow module window states resizable state works for a frameless window
  AssertionError [ERR_ASSERTION]: true == false
```

@zcbenz, can we merge as is and fix them later?

**Comment by timofonic** [comment]
Looking forward to this. Thanks a lot for your hard work! ;)

**Comment by zcbenz** [comment]
> @zcbenz, can we merge as is and fix them later?

I'm 👍 merging it and fixing the failing tests later, but as discussed in https://github.com/electron/maintainers-private/issues/68#issuecomment-319509251 we will cut a 1.7.x release first because merging this branch.

**Comment by alexeykuzmin** [comment]
> I'm 👍 merging it and fixing the failing tests later

That's just great.

Btw, I see [failed linux-x64 build](https://janky.githubapp.com/7499768/output), but I don't have access to the build logs, is everything ok there?
On my Linux x64 machine everything was fine.

**Comment by zcbenz** [comment]
> Btw, I see failed linux-x64 build, but I don't have access to the build logs, is everything ok there?
> On my Linux x64 machine everything was fine.

It was the same error with [travis ci](https://travis-ci.org/electron/electron/jobs/260162253), the tests on Linux CI machines were hanging after running a few tests, the stopped position was random.

It did not crash, otherwise the tests would be aborted and stack trace would show. And it was not a specific test that caused the hang, since the whole process was stalled and the it stopped after running random number of tests. 

The process tree in CI was:

```
root      59743  0.0  0.0 341756  4740 ?        Sl   Jul31   0:00      \_ docker-containerd-shim 4b3398c34060dd53ce6eacf0bc9fd74860b3ae87ff45ba5995b499d423507b6d /var/run/docker/libcontainerd/4b3398c34060dd53ce6eacf0bc9fd74860b3ae87ff45ba5995b499d423507b6d docker-runc
jenkins   59760  0.0  0.0  36184  9468 ?        Ss   Jul31   0:00      |   \_ python script/cibuild
jenkins   59798  0.0  0.0 314476 20056 ?        Sl   Jul31   0:00      |       \_ /usr/bin/Xvfb :99.0 -ac -screen 0 1024x768x16 +extension RANDR
jenkins   65784  0.0  0.0  37052 10336 ?        S    Jul31   0:00      |       \_ /usr/bin/python /workspace/electron/script/test.py --ci --rebuild_native_modules
jenkins   65952  0.0  0.1 1403316 178788 ?      Sl   Jul31   0:08      |       |   \_ /workspace/electron/out/D/electron --ci --rebuild_native_modules spec
jenkins   65954  0.0  0.0 578556 77528 ?        S    Jul31   0:00      |       |       \_ /workspace/electron/out/D/electron --type=zygote --no-sandbox
jenkins   65992  0.0  0.1 1353784 143360 ?      Sl   Jul31   0:03      |       |       |   \_ /workspace/electron/out/D/electron --type=renderer --js-flags=--expose_gc --no-sandbox --primordial-pipe-token=E0D510A5F5BA4B0B89EDE036C78FEAAB --lang=en-US --standard-schemes=app,zoom --secure-schemes=app,zoom --app-path=/workspace/electron/spec --node-integration=true --webview-tag=true --no-sandbox --enable-pinch --num-raster-threads=4 --enable-main-frame-before-activation --content-image-texture-target=0,0,3553;0,1,3553;0,2,3553;0,3,3553;0,4,3553;0,5,3553;0,6,3553;0,7,3553;0,8,3553;0,9,3553;0,10,3553;0,11,3553;0,12,355
jenkins   65983  0.0  0.0 840740 113908 ?       Sl   Jul31   0:01      |       |       \_ /workspace/electron/out/D/electron --type=gpu-process --no-sandbox --supports-dual-gpus=false --gpu-driver-bug-workarounds=7,24,76 --disable-gl-extensions=GL_KHR_blend_equation_advanced GL_KHR_blend_equation_advanced_coherent --disable-webrtc-hw-encoding --gpu-vendor-id=0x1a03 --gpu-device-id=0x2000 --gpu-driver-vendor --gpu-driver-version --gpu-driver-date --service-request-channel-token=DF88558AA5939BA2DA5915635025D10F
jenkins   66031  0.3  0.0 1427904 70216 ?       S    Jul31   6:57      |       |       \_ /workspace/electron/out/D/electron --ci --rebuild_native_modules spec
jenkins   65990  0.0  0.0  43596   824 ?        S    Jul31   0:00      |       \_ dbus-launch --autolaunch b9e75b1347434d78a95c03a2851eb58c --binary-syntax --close-stderr
jenkins   65991  0.0  0.0  42764  1044 ?        Ss   Jul31   0:00      |       \_ /usr/bin/dbus-daemon --fork --print-pid 5 --print-address 7 --session
```

The tests are running fine on local machines, and only hang when running on Docker. I can't find a way to fix this, it seems that the debug version of Chromium somehow does not work well on Docker. /cc @alespergl

**Comment by setpixel** [comment]
With all checks passing, when can we expect to see a 1.8 pre release?

**Comment by jkleinsc** [comment]
@setpixel I don't think everything is currently passing, but I have restarted the builds to verify that it wasn't because of a missing libcc build.

**Comment by setpixel** [comment]
@jkleinsc ah yes - I know they aren't currently passing. I'm just asking: once they are, how much longer after passing checks will there be a 1.8 release?

**Comment by jkleinsc** [comment]
@setpixel After everything is passing, I would expect to do a release within a day.

**Comment by alexeykuzmin** [comment]
Rebased onto the latest master and updated the libcc revision to the latest with Chromium 59.

**Comment by MarshallOfSound** [comment]
Do we have a new ABI value for this merge?  Assuming I'll need to update the electron section of https://github.com/lgeiger/node-abi

**Comment by alexeykuzmin** [comment]
> Do we have a new ABI value for this merge? Assuming I'll need to update the electron section of https://github.com/lgeiger/node-abi

I guess probably @kevinsawicki knows.

**Comment by setpixel** [comment]
@alexeykuzmin so close! did mas fail on a network error?

**Comment by alexeykuzmin** [comment]
> did mas fail on a network error?

@setpixel Yes, it did. I just restarted the builds to get them all green.

**Comment by setpixel** [comment]
With all green, can we expect 1.8 soon? 😄

File: spec/api-browser-window-spec.js [hunk 1]
```diff
@@ -1185,7 +1185,7 @@ describe('BrowserWindow module', function () {
         })
       })
 
-      it('can print to PDF', function (done) {
+      xit('can print to PDF', function (done) {
```
Comment (MarshallOfSound) [inline spec/api-browser-window-spec.js]: This should be noted somewhere that it was disabled and needs to be re-enabled before `1.8.x` can go stable

File: spec/api-browser-window-spec.js [hunk 2]
```diff
@@ -1959,7 +1962,7 @@ describe('BrowserWindow module', function () {
         assert.equal(w.isResizable(), true)
       })
 
-      it('works for a frameless window', () => {
+      xit('works for a frameless window', () => {
```
Comment (MarshallOfSound) [inline spec/api-browser-window-spec.js]: This should be noted somewhere that it was disabled and needs to be re-enabled before `1.8.x` can go stable

Comment (alexeykuzmin) [inline spec/api-browser-window-spec.js]: Right now it is mentioned in the PR's description. And I'm going to create an issue to fix the test if/after the PR is merged.

File: spec/api-browser-window-spec.js (hunk 1 above)
Comment (alexeykuzmin) [inline spec/api-browser-window-spec.js]: Right now it is mentioned in the PR's description. And I'm going to create an issue to fix the test if/after the PR is merged.

**Comment by setpixel** [comment]
Do you guys have any idea when you'll cut a 1.8 release? 😃

**Comment by jkleinsc** [comment]
@setpixel there are two failing Windows tests that need to be fixed before 1.8 is released.  Once we are clear on those issues we will do a 1.8 release.  We are tentatively planning on doing a release on Tuesday (August 22nd).

BTW those issues are #10279 and #10280.

**Comment by stanleyxu2005** [comment]
I am just curious that integration to foo version of libchroium cannot be fully automated with some CI?

**Comment by alexeykuzmin** [comment]
@stanleyxu2005 no, it can't be automated.

**Pull request merged**